10-inference.Rmd

```diff
@@ -0,0 +1,789 @@
+head(boot1)
+boot1_dist <- ggplot(boot1, aes(price)) + 
+    geom_histogram(fill="#0072B2", color="#e9ecef") +
+    xlab("Price per night ($)") 
+
+boot1_dist
+
+summarise(boot1, mean = mean(price))
+```
+
+Notice that our bootstrap distribution has a similar shape to the original
+sample distribution. Though the shapes of the distributions are similar, they
+are not identical. You'll also notice that the original sample mean and the
+bootstrap sample mean differ. How might that happen? Remember that we are
+sampling with replacement from the original sample, so we don't end up with the
+same sample values again. We are trying to mimic drawing another sample from
+the population without actually having to do that.
+
+Let's now take 1500 bootstrap samples from the original sample we drew from the
+population (`one_sample`)  using `rep_sample_n` and calculate the means for
+each of those replicates. Recall that this assumes that `one_sample` *looks like*
+our original population; but since we do not have access to the population itself,
+this is often the best we can do.
+
+```{r 11-bootstrapping4, echo = TRUE, message = FALSE, warning = FALSE}
+boot1500 <- one_sample %>%
+    rep_sample_n(size = 40, replace = TRUE, reps = 1500)
+head(boot1500)
+tail(boot1500)
+boot1500_means <- boot1500 %>% 
+  group_by(replicate) %>% 
+  summarize(mean = mean(price))
+head(boot1500_means)
+tail(boot1500_means)
+```
+
+We now calculate point estimates for each bootstrap sample and generate a bootstrap
+distribution of our point estimates. The bootstrap distribution suggests how we
+might expect our point estimate to behave if we took another sample.
+
+```{r 11-bootstrapping5, echo = TRUE, message = FALSE, warning = FALSE, fig.cap = "Distribution of the bootstrap sample means"}
+boot_est_dist <-  ggplot(boot1500_means, aes(x = mean)) +
+    geom_histogram(fill="#0072B2", color="#e9ecef") +
+    xlab("Sample mean price per night ($)") 
+```
+
+Let's compare our bootstrap distribution with the true sampling distribution (taking many samples from the population). 
+
+```{r 11-bootstrapping6, echo = F, message = FALSE, warning = FALSE, fig.cap = "Comparison of distribution of the bootstrap sample means and sampling distribution"}
+
+samples <- rep_sample_n(airbnb, size = 40, reps = 1500)
+
+sample_estimates <- samples %>% 
+    group_by(replicate) %>% 
+    summarise(sample_mean = mean(price))
+
+sampling_dist <-  ggplot(sample_estimates, aes(x = sample_mean)) +
+    geom_histogram(fill="#0072B2", color="#e9ecef") +
+    xlab("Sample mean price per night ($)")
+
+sampling_dist <- sampling_dist +
+  annotate("text", x = max_x(sampling_dist), max_count(sampling_dist), hjust = 1, vjust =1,
+           label = paste("mean = ", round(mean(sample_estimates$sample_mean), 1))) + 
+  annotate("text", x = max_x(sampling_dist), max_count(sampling_dist), hjust = 1, vjust =3,
+           label = paste("sd = ", round(sd(sample_estimates$sample_mean), 1))) + 
```

Comment:
I don't think we should annotate the standard deviation. Instead we can just eyeball the spread and say that it looks similar.